src/ext/java/org/opentripplanner/ext/emissions/Co2EmissionsDataReader.java

```diff
@@ -0,0 +1,141 @@
+package org.opentripplanner.ext.emissions;
+
+import com.csvreader.CsvReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.zip.ZipFile;
+import org.opentripplanner.framework.lang.Sandbox;
+import org.opentripplanner.graph_builder.issue.api.DataImportIssueStore;
+import org.opentripplanner.transit.model.framework.FeedScopedId;
+
+/**
+ * This class handles reading the CO₂ emissions data from the files in the GTFS package
+ * and saving it in a map.
+ */
+@Sandbox
+public class Co2EmissionsDataReader {
+
+  private final DataImportIssueStore issueStore;
+
+  public Co2EmissionsDataReader(DataImportIssueStore issueStore) {
+    this.issueStore = issueStore;
+  }
+
+  /**
+   * Read files in a GTFS directory.
+   * @param directory
+   * @return emissions data
+   */
+  public Map<FeedScopedId, Double> readGtfs(File directory) {
+    String feedId = "";
+    Map<FeedScopedId, Double> emissionsData = new HashMap<>();
+    try (InputStream feedInfoStream = new FileInputStream(directory + "/feed_info.txt")) {
+      feedId = readFeedId(feedInfoStream);
+    } catch (IOException e) {
+      issueStore.add("InvalidData", "Reading feed_info.txt failed.");
+    }
+    try (InputStream stream = new FileInputStream(directory + "/emissions.txt")) {
+      emissionsData = readEmissions(stream, feedId);
+    } catch (IOException e) {
+      issueStore.add("InvalidData", "Reading emissions.txt failed.");
+    }
+    return emissionsData;
+  }
+
+  /**
+   * Read files in a GTFS zip file.
+   * @param file
+   * @return emissions data
+   */
+  public Map<FeedScopedId, Double> readGtfsZip(File file) {
+    try {
+      ZipFile zipFile = new ZipFile(file, ZipFile.OPEN_READ);
+      String feedId = readFeedId(zipFile.getInputStream(zipFile.getEntry("feed_info.txt")));
+      InputStream stream = zipFile.getInputStream(zipFile.getEntry("emissions.txt"));
+      Map<FeedScopedId, Double> emissionsData = readEmissions(stream, feedId);
+      zipFile.close();
+      return emissionsData;
+    } catch (IOException e) {
+      issueStore.add("InvalidData", "Reading emissions data failed.");
```

Comment:
```suggestion
      issueStore.add("InvalidEmissionData", "Reading emissions data failed.");
```

Reply:
I think in these scenarios where the whole thing fails, we can also add a log line.